src/client/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -81,8 +81,8 @@ internal ClientApplicationBase(ApplicationConfiguration config)
             RequestContext requestContext,
             ITokenCacheInternal cache)
         {
-            var authority = Microsoft.Identity.Client.Instance.Authority.CreateAuthorityForRequest(
-               requestContext.ServiceBundle.Config.AuthorityInfo,
+            var authority = Instance.Authority.CreateAuthorityForRequest(
+               requestContext,
```

Comment:
It's better to send exactly the objects needed (in this case the InstanceDiscoveryManager) rather than sending the entire  context object.

Reply:
The instance discovery methods need the request context and since the instance discovery manager and the authority info is in the request context. I think it is better to just send the one object alone. This also reduces the number of parameters that need to be passed around.

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/4a0d46b37fb067163df88262ab3db473a81609c7/src/client/Microsoft.Identity.Client/Instance/Authority.cs#L224